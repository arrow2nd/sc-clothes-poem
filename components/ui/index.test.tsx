import { fireEvent, render } from '@testing-library/react'
import { act } from 'react-dom/test-utils'

import UI from './index'

describe('UI', () => {
  test('衣装名を指定して検索できるか', async () => {
    const { container, getAllByRole } = render(<UI poemText="" />)

    // 衣装名のコンボボックスを選択
    await act(async () => {
      const clothesCombobox = getAllByRole('combobox')[1]

      fireEvent.change(clothesCombobox, {
        target: { value: 'ほしあかり' }
      })

      await new Promise((r) => setTimeout(r, 100))

      fireEvent.keyDown(clothesCombobox, {
        key: 'Enter',
        code: 'Enter',
        charCode: 13
      })
    })

    expect(container).toMatchSnapshot()
  })
})
