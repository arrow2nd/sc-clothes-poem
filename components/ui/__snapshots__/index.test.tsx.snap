// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI 衣装名を指定して検索できるか 1`] = `
<div>
  <div
    class="flex-grow mx-6 md:mx-12"
  >
    <div
      class="flex justify-center mb-16"
    >
      <div
        class="w-full lg:w-auto max-w-lg lg:max-w-none"
      >
        <div
          class="flex flex-row items-center ml-2 mb-4 text-2xl text-main"
        >
          <svg
            class="mr-2"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
          <span
            class="tracking-widest"
          >
            Search
          </span>
        </div>
        <div
          class="flex flex-wrap"
        >
          <div
            class="w-full lg:w-72 m-2"
          >
            <div
              class="flex items-center rounded-md border-2 border-main text-main bg-white"
            >
              <input
                class="w-full h-9 ml-2.5 placeholder-sub rounded-md focus:outline-none"
                placeholder="ポエムの一部から"
                type="text"
              />
              <button
                class="mx-1 p-2 transition-colors focus:outline-none"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="w-full lg:w-72 m-2 bg-white css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-アイドル名から-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-1ycg57x-control"
            >
              <div
                class=" css-319lph-ValueContainer"
              >
                <div
                  class=" css-1yaegve-placeholder"
                  id="react-select-アイドル名から-placeholder"
                >
                  <p>
                    アイドル名から
                  </p>
                </div>
                <div
                  class=" css-17k8k4u-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-controls="react-select-アイドル名から-listbox"
                    aria-describedby="react-select-アイドル名から-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-owns="react-select-アイドル名から-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-アイドル名から-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-p70wra-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-lp1szy-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="w-full lg:w-72 m-2 bg-white css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-衣装名から-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-1ycg57x-control"
            >
              <div
                class=" css-319lph-ValueContainer"
              >
                <div
                  class=" css-354nsk-singleValue"
                >
                  ほしあかり
                </div>
                <div
                  class=" css-17k8k4u-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-controls="react-select-衣装名から-listbox"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-owns="react-select-衣装名から-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-衣装名から-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 0; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-p70wra-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-lp1szy-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full mb-16 border-b-2 border-main border-dashed"
    />
    <div
      class="mb-16 font-kiwi"
    >
      <div
        class="flex flex-wrap justify-center"
      >
        <div
          class="relative flex items-center w-96 h-60 m-2 p-8 border-2 border-main rounded-md bg-white text-main"
        >
          <div>
            <div
              class="mb-4 text-lg md:text-xl"
            >
              <p>
                しろはえの浴衣。
              </p>
              <p>
                赤い実のように甘くて楽しい時間
              </p>
            </div>
            <div
              class="mb-4 py-0.5 w-10 rounded-full border border-gray-100"
              style="background-color: rgb(255, 186, 214);"
            />
            <div
              class=""
            >
              <p
                class="mb-1 text-sm md:text-base"
              >
                ほしあかり
              </p>
              <p
                class="text-xs md:text-sm"
              >
                櫻木真乃
              </p>
            </div>
          </div>
          <div
            class="absolute right-0 bottom-0 p-8 flex flex-row space-x-4 text-main"
          >
            <a
              class="text-xl hover:text-black transition-colors"
              href="https://twitter.com/intent/tweet?text=%E3%81%97%E3%82%8D%E3%81%AF%E3%81%88%E3%81%AE%E6%B5%B4%E8%A1%A3%E3%80%82%0A%E8%B5%A4%E3%81%84%E5%AE%9F%E3%81%AE%E3%82%88%E3%81%86%E3%81%AB%E7%94%98%E3%81%8F%E3%81%A6%E6%A5%BD%E3%81%97%E3%81%84%E6%99%82%E9%96%93%0A%23%E3%82%B7%E3%83%A3%E3%83%8B%E3%83%9E%E3%82%B9%20%23%E3%81%BB%E3%81%97%E3%81%82%E3%81%8B%E3%82%8A%20%23%E6%AB%BB%E6%9C%A8%E7%9C%9F%E4%B9%83%0Ahttps%3A%2F%2Fshiny-poems.vercel.app%3Fid%3DHoshiAkari_SakuragiMano"
              rel="noopener noreferrer"
              target="_blank"
              title="ツイートする"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
                />
              </svg>
            </a>
            <button
              class="text-xl hover:text-black transition-colors focus:outline-none"
              title="コピーする"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="relative flex items-center w-96 h-60 m-2 p-8 border-2 border-main rounded-md bg-white text-main"
        >
          <div>
            <div
              class="mb-4 text-lg md:text-xl"
            >
              <p>
                しろはえの浴衣。
              </p>
              <p>
                跳ねる気持ちを浅縹は知っている
              </p>
            </div>
            <div
              class="mb-4 py-0.5 w-10 rounded-full border border-gray-100"
              style="background-color: rgb(20, 67, 132);"
            />
            <div
              class=""
            >
              <p
                class="mb-1 text-sm md:text-base"
              >
                ほしあかり
              </p>
              <p
                class="text-xs md:text-sm"
              >
                風野灯織
              </p>
            </div>
          </div>
          <div
            class="absolute right-0 bottom-0 p-8 flex flex-row space-x-4 text-main"
          >
            <a
              class="text-xl hover:text-black transition-colors"
              href="https://twitter.com/intent/tweet?text=%E3%81%97%E3%82%8D%E3%81%AF%E3%81%88%E3%81%AE%E6%B5%B4%E8%A1%A3%E3%80%82%0A%E8%B7%B3%E3%81%AD%E3%82%8B%E6%B0%97%E6%8C%81%E3%81%A1%E3%82%92%E6%B5%85%E7%B8%B9%E3%81%AF%E7%9F%A5%E3%81%A3%E3%81%A6%E3%81%84%E3%82%8B%0A%23%E3%82%B7%E3%83%A3%E3%83%8B%E3%83%9E%E3%82%B9%20%23%E3%81%BB%E3%81%97%E3%81%82%E3%81%8B%E3%82%8A%20%23%E9%A2%A8%E9%87%8E%E7%81%AF%E7%B9%94%0Ahttps%3A%2F%2Fshiny-poems.vercel.app%3Fid%3DHoshiAkari_KazanoHiori"
              rel="noopener noreferrer"
              target="_blank"
              title="ツイートする"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
                />
              </svg>
            </a>
            <button
              class="text-xl hover:text-black transition-colors focus:outline-none"
              title="コピーする"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="relative flex items-center w-96 h-60 m-2 p-8 border-2 border-main rounded-md bg-white text-main"
        >
          <div>
            <div
              class="mb-4 text-lg md:text-xl"
            >
              <p>
                しろはえの浴衣。
              </p>
              <p>
                差し色は赤い実と浅縹を
              </p>
            </div>
            <div
              class="mb-4 py-0.5 w-10 rounded-full border border-gray-100"
              style="background-color: rgb(255, 224, 18);"
            />
            <div
              class=""
            >
              <p
                class="mb-1 text-sm md:text-base"
              >
                ほしあかり
              </p>
              <p
                class="text-xs md:text-sm"
              >
                八宮めぐる
              </p>
            </div>
          </div>
          <div
            class="absolute right-0 bottom-0 p-8 flex flex-row space-x-4 text-main"
          >
            <a
              class="text-xl hover:text-black transition-colors"
              href="https://twitter.com/intent/tweet?text=%E3%81%97%E3%82%8D%E3%81%AF%E3%81%88%E3%81%AE%E6%B5%B4%E8%A1%A3%E3%80%82%0A%E5%B7%AE%E3%81%97%E8%89%B2%E3%81%AF%E8%B5%A4%E3%81%84%E5%AE%9F%E3%81%A8%E6%B5%85%E7%B8%B9%E3%82%92%0A%23%E3%82%B7%E3%83%A3%E3%83%8B%E3%83%9E%E3%82%B9%20%23%E3%81%BB%E3%81%97%E3%81%82%E3%81%8B%E3%82%8A%20%23%E5%85%AB%E5%AE%AE%E3%82%81%E3%81%90%E3%82%8B%0Ahttps%3A%2F%2Fshiny-poems.vercel.app%3Fid%3DHoshiAkari_HachimiyaMeguru"
              rel="noopener noreferrer"
              target="_blank"
              title="ツイートする"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M23 3a10.9 10.9 0 0 1-3.14 1.53 4.48 4.48 0 0 0-7.86 3v1A10.66 10.66 0 0 1 3 4s-4 9 5 13a11.64 11.64 0 0 1-7 2c9 5 20 0 20-11.5a4.5 4.5 0 0 0-.08-.83A7.72 7.72 0 0 0 23 3z"
                />
              </svg>
            </a>
            <button
              class="text-xl hover:text-black transition-colors focus:outline-none"
              title="コピーする"
            >
              <svg
                fill="none"
                height="1em"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="13"
                  rx="2"
                  ry="2"
                  width="13"
                  x="9"
                  y="9"
                />
                <path
                  d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
